pynicotine/downloads.py

```diff
@@ -543,6 +545,10 @@ def _upload_denied(self, msg):
                 self.get_file(username, virtual_path, transfer=download)
                 break
 
+            if (reason in {"Too many files", "Too many megabytes"} or reason.startswith("User limit of")
+                    and username not in self.limited_user_times):
+                self.limited_user_times[username] = time.monotonic()
```

Comment:
Consider creating `events.schedule()` for each username instead of time for the value.

Reply:
Since we'd have to iterate through the list of downloads for every limited user, I prefer keeping the current implementation.

Reply:
Then again, we likely won't have many limited users. I'll investigate.

Reply:
Ideally it would be best to avoid any batch operations that have the potential to trigger a large iteration.

There is no rush to retry these failed downloads, since we have a window of 12 or 5 minutes to do this, so each pending username auto-retry might as well be spread out across that time.

Reply:
Changed it to your initial suggestion.